model_compression_toolkit/core/keras/graph_substitutions/substitutions/activation_decomposition.py

```diff
@@ -70,18 +70,30 @@ def substitute(self,
 
         activation_node_name = op2d_node.name + '_post_activation'
 
-        activation_fw_attr = {
-            LAYER_NAME: activation_node_name,
-            TRAINABLE: False,
-            DATA_TYPE: FLOAT_32,
-            ACTIVATION: op2d_node.framework_attr.get(ACTIVATION)}
-
-        activation_node = common.graph.BaseNode(activation_node_name,
-                                                activation_fw_attr,
-                                                op2d_node.output_shape,
-                                                op2d_node.output_shape,
-                                                {},
-                                                Activation)
+        # Softmax is a special case where we need to know the default axis parameter used
+        # (for the distance metric in MP for KL-Divergence), and for this reason we create
```

Comment:
I would remove the cause reference (MP and KL) from here, it can be confusing if you just looked at this file without context.